model.go

```diff
@@ -72,15 +74,33 @@ func ModelFields(model Model) field.Names {
 	if modelType.Elem().Kind() != reflect.Struct {
 		panic("Expected Model to be a Ptr to a Struct")
 	}
+	fields := make(field.Names, 0)
 
-	modelValue := reflect.ValueOf(model).Elem()
+	fields = append(fields, modelFields(model)...)
 
+	return fields
+}
+
+func modelFields(model interface{}) field.Names {
 	fields := make(field.Names, 0)
 
-	for i := 0; i < modelValue.NumField(); i++ {
-		_field := modelValue.Field(i)
-		if _field.CanAddr() == true && _field.Addr().Type().Implements(fieldType) == true {
-			fields = append(fields, field.Name(modelType.Elem().Field(i).Name))
+	ifv := reflect.ValueOf(model)
+	if ifv.Kind() == reflect.Ptr {
+		ifv = ifv.Elem()
+	}
+
+	itf := reflect.TypeOf(model)
+	if itf.Kind() == reflect.Ptr {
+		itf = itf.Elem()
+	}
+
+	for i := 0; i < itf.NumField(); i++ {
+		v := ifv.Field(i)
+		t := itf.Field(i)
+		if v.CanAddr() == true && v.Addr().Type().Implements(fieldType) == true {
+			fields = append(fields, field.Name(itf.Field(i).Name))
+		} else if t.Anonymous == true && v.CanAddr() && v.Kind() == reflect.Struct {
+			fields = append(fields, modelFields(v.Addr().Interface())...)
 		}
 	}
```

Comment:
A short comment for `ivf` and `itf` would be beneficial.

Is there a reason for using `v.CanAddr() == true` in one place `v.CanAddr()` in another?  Same for any of the other == true checks here.